tests/logging-utils/test_compat_with_typing_Generic.py

```diff
@@ -1,16 +1,18 @@
 import logging
-from typing import GenericMeta, TypeVar
+import sys
 
-from eth_utils import HasLoggerMeta
+import pytest
 
-T = TypeVar("T")
+from eth_utils import HasLoggerMeta
 
 
-class HasLoggerCompatWithGeneric(metaclass=HasLoggerMeta.meta_compat(GenericMeta)):
-    ...
+@pytest.mark.skipif(sys.version_info[:2] == (3, 7), reason="GenericMeta not present in Python 3.7+")
+def test_has_logger_compat_with_typing_Generic():
+    from typing import GenericMeta
 
+    class HasLoggerCompatWithGeneric(metaclass=HasLoggerMeta.meta_compat(GenericMeta)):
```

Comment:
Does this mean that the whole approach of using `meta_compat` is broken for `Generic` in 3.7+? Do we need a different approach?

Reply:
Not something that needs to be fixed in this library I don't think but it would look something like this.

```python
if is_python_36():
    from typing import GenericMeta, Generic
    class GenericWithLogger(Generic, HasLoggerMeta.meta_compat(GenericMeta)):
        ...
elif is_python_37_or_greater():
    from typing import Generic
    class GenericWithLogger(HasLogger, Generic):
        ...
```

Basically, `typing.Generic` is no longer implemented with a metaclass in 3.7 so there's no need to do the compat layer.